libnetwork/iptables/firewalld.go

```diff
@@ -52,11 +58,43 @@ func firewalldInit() error {
 	if connection, err = newConnection(); err != nil {
 		return fmt.Errorf("Failed to connect to D-Bus system bus: %v", err)
 	}
-	firewalldRunning = checkRunning()
+
+	firewalldRunning, err = checkRunning()
+
+	if err != nil {
+		return err
+	}
+	// If firewalld is not running on the first check poll
 	if !firewalldRunning {
-		connection.sysconn.Close()
-		connection = nil
+		tick := time.NewTicker(500 * time.Millisecond)
+		timeLapse := time.Duration(0)
+		defer tick.Stop()
+
+	TickLoop:
+		for {
+			select {
+			case <-tick.C:
+				firewalldRunning, _ = checkRunning()
+				timeLapse += 500 * time.Millisecond
+
+				log.G(context.TODO()).Debugf("Checking status of Firewalld")
+
+				if firewalldRunning || timeLapse >= FireWalldInitializationTimeout {
```

Comment:
Wondering why the manual `timeLapse`; would `time.Since()` (https://pkg.go.dev/time#Since) work for this? (i.e. set a `startTime` before going into the loop, and compare against that).